### ISSUE TITLE
ceph: enable to remove an arbitrary OSD in PVC-based cluster

### ISSUE THREAD

**Issue by satoru-takeuchi** [rook/rook#6948]
**Is this a bug report or feature request?**
* Feature Request

**What should the feature do:**

Currently, we can only remove the OSD which corresponds to the PVC having the highest index.
https://github.com/rook/rook/blob/master/Documentation/ceph-osd-mgmt.md#remove-an-osd-from-a-pvc

It's inconvenient especially when we need to remove an unhealthy OSD.

**What is use case behind this feature:**

- Remove arbitrary unhealthy OSDs
- Remove OSDs in a node when retiring this node. It's essential in non-portable PVs.

**Comment by satoru-takeuchi** [comment]
I found an unofficial workaround that replaces two PVC-backed OSDs without decreasing `count` field.

```console
NS=rook-ceph
CEPHCMD="kubectl -n $NS exec rook-ceph-toolbox-XXX -- "
$CEPHCMD  osd out $OSD_ID
kubectl -n $NS delete pvc $PVC_NAME --wait=false
kubectl -n $NS delete deployment $OSD_DEPLOYMENT
kubectl -n $NS delete secrets $OSD_AUTH
kubectl -n $NS delete job $OSD_PREPARE_JOB
$CEPHCMD auth del osd.$OSD_ID
$CEPHCMD osd purge $OSD_ID --yes-i-really-mean-it
kubectl -n $NS delete pod $(kubectl -n $NS get pod -lapp=rook-ceph-operator -ojsonpath={'.items[0].metadata.name}')
```

I consider at least one of the following two changes is necessary.

- Describe the above-mentioned steps in the ceph-osd-mgmt.md.
- Provide the way to remove an arbitrary OSD removal.

Eventually, it would be better to do both changes.

**Comment by travisn** [comment]
This is the same root cause as #6287. Basically, we would need to use labels to associate the PVCs with a device set instead of a naming convention.

**Comment by satoru-takeuchi** [comment]
So we need at least the following two changes.

- Provide the way to specify pvc's label in removing OSD.
- Allow holes in the index of pvc name.

**Comment by travisn** [comment]
> So we need at least the following two changes.
> 
> * Provide the way to specify pvc's label in removing OSD.

When removing an OSD I don't know if we need to specify a label. If you know the PVC by name, you can just delete it (the osd purge job will also find it and purge it by name).

> * Allow holes in the index of pvc name.

Yes, this is a code change for the operator to use labels instead of exactly generated PVC names. I was looking at this a few months ago, I'll take another look at it soon...

**Comment by satoru-takeuchi** [comment]
Resolved in #6982.
